File: 7-kyu/next-prime/python/solution.py
def is_prime(n):
    if n < 2:
        return False
    if n == 2 or n == 3 or n == 5:
        return True
    if n % 2 == 0 or n % 3 == 0 or n % 5 == 0:
        return False
    for i in range(5, int(n ** 0.5) + 1, 2):
        if n % i == 0:
            return False
    return True

def next_prime(n):
    if is_prime(n):
        n += 1
    while not is_prime(n):
        n += 1
    return n

File: 7-kyu/next-prime/javascript/solution.js
const isPrime = n => {
  if (n < 2) {
    return false;
  }
  if (n === 2 || n === 3) {
    return true;
  }
  if (n % 2 === 0 || n % 3 === 0) {
    return false;
  }
  const limit = Math.floor(Math.sqrt(n)) + 1;
  for (let i = 3; i < limit; i += 2) {
    if (n % i === 0){
      return false;
    }
  }
  return true;
};

const nextPrime = n => {
  if (isPrime(n)) {
    n++;
  }
  while (!isPrime(n)) {
    n++;
  }
  return n;
};
